### ISSUE TITLE
"Blocked by rule: temporary_blocked2"

### ISSUE THREAD

**Issue by adbourdages** [playgameservices/play-games-plugin-for-unity#2797]
Hello,

I have a game that has long been in production (I mean to say the GPGS implementation works well, most of the time.) A user contacted me about his leaderboard scores not updating and decided to investigate.

It turns out that 2 of the 9 leaderboards in the game give the error "Blocked by rule: temporary_blocked2" under the following conditions:
- only those 2 leaderboards
- the first 25 scores of All/Week/24h are shown properly (in the native GPGS leaderboard display)
- If you scroll down past the 25 scores, the app tries to load more data (get the spinning wheel and all)
- Then it stops and report that "there is no internet connection" which is not correct
- Looking at the logs, I see the error: "Blocked by rule: temporary_blocked2"

The full error stack is below. A google search on the error returns some old stuff about Geocoder only - nothing Play related.

Halp!

`2019-12-02 10:14:22.332 2799-16982/? W/LeaderboardAgent: Failed to retrieve leaderboard scores for 121939725521 CgkI0ZmyocYDEAIQoQE DAILY
    com.android.volley.ServerError
        at com.android.volley.toolbox.BasicNetwork.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):44)
        at sps.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):17)
        at com.android.volley.NetworkDispatcher.a(:com.google.android.gms@19629037@19.6.29 (120400-278422107):6)
        at com.android.volley.NetworkDispatcher.run(:com.google.android.gms@19629037@19.6.29 (120400-278422107):2)
2019-12-02 10:14:22.332 2799-16982/? W/LeaderboardAgent: {"code":500}
2019-12-02 10:14:22.380 2340-4224/? I/GCM.HTTP: Last HTTP status code for blocked entry: 500
2019-12-02 10:14:22.380 2340-4224/? W/GCM.HTTP: HTTP request blocked due to automatic backoff: https://www.googleapis.com/games/v1/leaderboards/CgkI0ZmyocYDEAIQoQE/scores/PUBLIC?timeSpan=DAILY&language=en-CA&maxResults=25&pageToken=AL7ALK1Vvg3DNoN56Vewje_hTtKPaJ3RkquASZYPCBo3KVBhjF2WG-9nfxUNNwCzf_el-SdjMPc30ntsEWIxbu5pANBhwmvy1Ps7GrxSTK8LA-bNur-hLmDqHJPILFk0s31gqPIIl-tWAl6v7aVsR91jLYk53fHxPUs7Hs8vrc9sPLEH4qTyh9w4SrsNwd1s6EtTsQUz4eUEacwp
2019-12-02 10:14:22.381 2799-6658/? W/GoogleURLConnFactory: Blocked by temporary_blocked2: https://www.googleapis.com/games/v1/leaderboards/CgkI0ZmyocYDEAIQoQE/scores/PUBLIC?timeSpan=DAILY&language=en-CA&maxResults=25&pageToken=AL7ALK1Vvg3DNoN56Vewje_hTtKPaJ3RkquASZYPCBo3KVBhjF2WG-9nfxUNNwCzf_el-SdjMPc30ntsEWIxbu5pANBhwmvy1Ps7GrxSTK8LA-bNur-hLmDqHJPILFk0s31gqPIIl-tWAl6v7aVsR91jLYk53fHxPUs7Hs8vrc9sPLEH4qTyh9w4SrsNwd1s6EtTsQUz4eUEacwp
2019-12-02 10:14:22.382 2799-16982/? W/LeaderboardAgent: Failed to retrieve leaderboard scores for 121939725521 CgkI0ZmyocYDEAIQoQE DAILY
    com.android.volley.NoConnectionError: abzx: Blocked by rule: temporary_blocked2
        at com.android.volley.toolbox.BasicNetwork.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):46)
        at sps.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):17)
        at com.android.volley.NetworkDispatcher.a(:com.google.android.gms@19629037@19.6.29 (120400-278422107):6)
        at com.android.volley.NetworkDispatcher.run(:com.google.android.gms@19629037@19.6.29 (120400-278422107):2)
     Caused by: abzx: Blocked by rule: temporary_blocked2
        at acae.a(:com.google.android.gms@19629037@19.6.29 (120400-278422107):5)
        at abzu.a(:com.google.android.gms@19629037@19.6.29 (120400-278422107):6)
        at abzz.a(Unknown Source:2)
        at abzu.a(:com.google.android.gms@19629037@19.6.29 (120400-278422107):20)
        at abzu.execute(:com.google.android.gms@19629037@19.6.29 (120400-278422107):8)
        at abzu.execute(:com.google.android.gms@19629037@19.6.29 (120400-278422107):7)
        at com.android.volley.toolbox.HttpClientStack.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):18)
        at spr.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):1)
        at bon.executeRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):1)
        at com.android.volley.toolbox.BasicNetwork.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):10)
        at sps.performRequest(:com.google.android.gms@19629037@19.6.29 (120400-278422107):17) 
        at com.android.volley.NetworkDispatcher.a(:com.google.android.gms@19629037@19.6.29 (120400-278422107):6) 
        at com.android.volley.NetworkDispatcher.run(:com.google.android.gms@19629037@19.6.29 (120400-278422107):2) `

**Comment by ozdemir08** [comment]
Hi, are you still having the same issue?

**Comment by adbourdages** [comment]
I assume I still do since I haven't changed anything. But when checking on the one test phone I currently have with me, I'm getting other errors rights now :-/ I can't say whether they are related or not but the behaviour is somewhat similar: the Play UI can load the first 25 scores but when scrolling it attempts to load more scores and that's where it stops working.

```
2020-02-21 12:26:56.125 32285-32521/? E/Volley: [1446] BasicNetwork.performRequest: Unexpected response code 500 for https://www.googleapis.com/games/v1/leaderboards/<removed>/scores/PUBLIC?timeSpan=WEEKLY&language=en-CA&maxResults=25&pageToken=<removed>
2020-02-21 12:26:56.126 32285-21606/? W/LeaderboardAgent: Failed to retrieve leaderboard scores for 121939725521 <LEADERBOARDID> WEEKLY
    com.android.volley.ServerError
        at com.android.volley.toolbox.BasicNetwork.performRequest(:com.google.android.gms@200414037@20.04.14 (120400-294335909):44)
        at sgu.performRequest(:com.google.android.gms@200414037@20.04.14 (120400-294335909):17)
        at com.android.volley.NetworkDispatcher.a(:com.google.android.gms@200414037@20.04.14 (120400-294335909):6)
        at com.android.volley.NetworkDispatcher.run(:com.google.android.gms@200414037@20.04.14 (120400-294335909):2)
```

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
